### PR TITLE
Us 01 api

### PR DESCRIPTION
**Pull request by Broom94**

### CODE REVIEW

**Comment by weesteeky** [comment]
pending TA Approval

**Comment by jamescrone** [comment]
fix merge conflicts in package.json

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+/typings
+
+# System Files
+.DS_Store
+Thumbs.db
+
+
+/package-lock.json
+
+node_modules
```
Comment (matthewdoreilly) [inline .gitignore]: I'd tidy up this gitignore file. Not all the files/folders specified here will ever be included in your project.

File: README.md [hunk 2]
```diff
@@ -1 +1,12 @@
...
+Express setup:
+-npm install express
+-npm install mysql
+
+Run: node app.js
+
+Angular:
+-npm install angular
+
+Run: npm start
```
Comment (matthewdoreilly) [inline README.md]: 👍

File: SQLstructure.sql [hunk 3]
```diff
@@ -0,0 +1,16 @@
+DROP DATABASE IF EXISTS klattice;
```
Comment (matthewdoreilly) [inline SQLstructure.sql]: What DB user is using this DB?

File: api/.gitignore [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+testem.log
+/typings
+
+# System Files
+.DS_Store
+Thumbs.db
+
+
+/package-lock.json
+api/package-lock.json
```
Comment (matthewdoreilly) [inline api/.gitignore]: Same comment as above.

> I'd tidy up this gitignore file. Not all the files/folders specified here will ever be included in your project.

File: api/app.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+const express = require('express')
+const app = express()
+const db = require('./db')
+
+app.get('/', function(req, res) {
+    res.send('<h1>First message from expreess</h1>\n');
+    console.log('Request processed');
+});
```
Comment (matthewdoreilly) [inline api/app.js]: Yous can delete this GET method, it's not really needed.

File: klattice/.gitignore [hunk 6]
```diff
@@ -44,3 +44,5 @@ testem.log
 # System Files
 .DS_Store
 Thumbs.db
+
+/package-lock.json
```
Comment (matthewdoreilly) [inline klattice/.gitignore]: You shouldn't ignore the `package-lock.json` file. You should push it up, so when all the devs in the team clone/pull your repo, everyone will be using (locked down to) the same version of each dependency.

File: SQLstructure.sql [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+CREATE DATABASE klattice;
+USE klattice;
+
+CREATE TABLE Role (
```
Comment (matthewdoreilly) [inline SQLstructure.sql]: Have DB, table and column names lowercase, and words separated by underscores. Good convention to follow when writing SQL.

File: api/db.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+const db = mysql.createConnection({
+    host: 'localhost',
+    user: 'username',
+    password: 'password',
+    database: 'klattice'
+});
+
+exports.getRoles = function (callback) {
+    db.query(
+        "SELECT roleID, roleName FROM Role;",
```
Comment (matthewdoreilly) [inline api/db.js]: Same comment as above.

> Have DB, table and column names lowercase, and words separated by underscores. Good convention to follow when writing SQL.

File: klattice/src/app/app.component.ts [hunk 9]
```diff
@@ -1,10 +1,43 @@
...
+//   },
+//   {
+//     roleID: 3,
+//     roleName: 'CEO'
+//   },
+//   {
+//     roleID: 4,
+//     roleName: 'Water boy'
+//   }
+// ];
```
Comment (matthewdoreilly) [inline klattice/src/app/app.component.ts]: Delete the whitespace here, and delete commented out code.

File: SQLstructure.sql (hunk 3 above)
Comment (Broom94) [reply SQLstructure.sql]: do you want the user details in this?

File: api/app.js (hunk 5 above)
Comment (Broom94) [reply api/app.js]: done

File: api/db.js (hunk 8 above)
Comment (Broom94) [reply api/db.js]: done

File: SQLstructure.sql (hunk 7 above)
Comment (Broom94) [reply SQLstructure.sql]: done

File: klattice/src/app/app.component.ts (hunk 9 above)
Comment (Broom94) [reply klattice/src/app/app.component.ts]: done

File: klattice/.gitignore (hunk 6 above)
Comment (Broom94) [reply klattice/.gitignore]: done

File: api/.gitignore (hunk 4 above)
Comment (Broom94) [reply api/.gitignore]: done

File: .gitignore (hunk 1 above)
Comment (Broom94) [reply .gitignore]: done

File: klattice/.gitignore [hunk 10]
```diff
@@ -31,16 +11,8 @@ speed-measure-plugin.json
...
-npm-debug.log
-yarn-error.log
-testem.log
-/typings
 
 # System Files
 .DS_Store
-Thumbs.db
+
+
```
Comment (matthewdoreilly) [inline klattice/.gitignore]: Delete this unnecessary whitespace here
